app/models/validations/restaurant_name_validator.rb

```diff
@@ -0,0 +1,11 @@
+class NameValidator < ActiveModel::Validator
```

Comment:
Great for practice 👍 

Just for the general knowledge, this can be achieved by defining in the model
`validates :name, presence: true, uniqueness: true
`

Reply:
I couldn't, because I wanted to controller the status called returned by this and I had to add a different error